### ISSUE TITLE
Allow _click of MenuItem in a MenuBar

### ISSUE THREAD

**Issue by eiswind** [mvysny/karibu-testing#33]
Currently MenuItem._click() seems to be tied to a contextMenu.
If the parent is a menubar I guess it could fire a simple click event. (?)

**Comment by mvysny** [comment]
Yeah, that sounds pretty reasonable, thanks! It seems that the ContextMenu is there just to verify that the menu item is actually visible and enabled. I wonder whether we can do the same for the MenuBar...

**Comment by mvysny** [comment]
Alternatively I can implement `menuBar._click(MenuItem)`.

**Comment by eiswind** [comment]
That way one would need to lookup both components. I worked around with a helper
```
_get<MenuItem> { text = "Logfiles" }._menuBarClick()
```

**Comment by eiswind** [comment]
> Alternatively I can implement `menuBar._click(MenuItem)`.

That would help in some places!

**Comment by mvysny** [comment]
> _get<MenuItem> { text = "Logfiles" }._menuBarClick()

Awesome workaround :-D Yet I'm hesitant to include this into Karibu-Testing since it's exactly as you said - no checks are done. Actually `_get()` performs the visibility checks, but doesn't check the enablement of the menu bar.

> menuBar._click(MenuItem)

Thanks! This has been implemented in the commit #53fb104 and will be released with the next release of Karibu-Testing :+1: 

Now that the workarounds are in place, let's wait for Vaadin folk to fix https://github.com/vaadin/vaadin-menu-bar-flow/issues/83 so that I can implement the `_click()` method properly.

**Comment by mvysny** [comment]
Resolved — clicking MenuBar items is fully supported. Since `MenuBar` implements `HasMenuItems`, you click items **on the `MenuBar` you hold** (unambiguous — no searching for "the" MenuBar):

```kotlin
menuBar._clickItemWithCaption("foo")   // by caption
menuBar._clickItemWithID("foo")        // by id
menuBar._clickItemWithIcon(icon)       // by icon
menuBar._click(menuItem)               // by MenuItem reference
```

These handle nested sub-menu items and `isChecked` toggling, and perform the usual visibility/enabled checks. Covered by `MenuBarTest`.

This mirrors the `ContextMenu` reference API (`contextMenu._clickItemWithCaption(...)`), which is the right parallel: a `MenuBar` is a first-class component you locate directly (`_get<MenuBar>()`), unlike a `ContextMenu` that's hidden behind a target component (for that case see the target-based API added in #20, e.g. `component._clickContextMenuItemWithCaption(...)`).

The one thing deliberately **not** supported is the no-arg `MenuItem._click()` called directly on a MenuBar item: a MenuBar item has no public back-reference to its `MenuBar` (`getContextMenu()` is `null`, `getParent()` is empty), so resolving it would require guessing the owning MenuBar by scanning the UI — which picks the wrong one when an app has more than one MenuBar (or only one happens to be visible). Rather than guess, `MenuItem._click()` fails fast with a message pointing to the methods above. Use `menuBar._click(menuItem)` when you hold a `MenuItem` reference.

Closing as resolved.
